src/filters/oidc/jwks_resolver.cc

```diff
@@ -15,20 +15,23 @@ DynamicJwksResolverImpl::JwksFetcher::JwksFetcher(
       ioc_(io_context),
       periodic_fetch_interval_sec_(periodic_fetch_interval_sec),
       timer_(ioc_, periodic_fetch_interval_sec_) {
-  timer_.async_wait(
-      [this](const boost::system::error_code& ec) { this->request(ec); });
+  // Extract initial JWKs.
+  // After timer callback sucessful, next timer invocation will be scheduled.
+  boost::system::error_code ec;
+  this->request(ec);
 }
 
 void DynamicJwksResolverImpl::JwksFetcher::request(
     const boost::system::error_code&) {
   boost::asio::spawn(ioc_, [this](boost::asio::yield_context yield) {
     auto resp = http_ptr_->Get(jwks_uri_, {}, "", "", "", ioc_, yield);
+    auto next_schedule_interval = periodic_fetch_interval_sec_;
 
     if (resp == nullptr) {
-      spdlog::info("{}: HTTP connection error", __func__);
+      spdlog::warn("{}: HTTP connection error", __func__);
     } else {
       if (resp->result() != boost::beast::http::status::ok) {
-        spdlog::info("{}: HTTP response error: {}", __func__,
+        spdlog::warn("{}: HTTP response error: {}", __func__,
```

Comment:
this can be "error" level?

Reply:
The reason is same as https://github.com/istio-ecosystem/authservice/pull/160#discussion_r715548073